mne_bids/tests/test_write.py

```diff
@@ -401,7 +401,9 @@ def test_fif(_bids_validate, tmpdir):
     ch_pos = dict(zip(ch_names, elec_locs))
     meg_montage = mne.channels.make_dig_montage(ch_pos=ch_pos,
                                                 coord_frame='head')
-    raw.set_montage(meg_montage)
+
+    with pytest.warns(RuntimeWarning, match='Not setting position'):
+        raw.set_montage(meg_montage)
```

Comment:
I am not so sure with swallowing the warning here. The warning that is being raised here seems odd to me: https://github.com/mne-tools/mne-bids/runs/2817614140#step:12:391

> RuntimeWarning: Not setting positions of 316 eeg/grad/mag/stim channels found in montage
> Consider setting the channel types to be of EEG/sEEG/ECoG/DBS using inst.set_channel_types before calling inst.set_montage.

cc @larsoner

Reply:
The warning is meant to tell you that your `set_montage` call is not setting the positions of some data channels, which is more likely to be an error of some sort than it is to be intentional. Is it intentional here?

Reply:
I find it a bit surprising that the positions are not being set, this is what we are doing (for the sake of the test) before calling `raw.set_montage(meg_montage)`:

```python
    raw = _read_raw_fif(raw_fname)
    # add data in as a montage for MEG
    ch_names = raw.ch_names
    elec_locs = np.random.random((len(ch_names), 3)).tolist()
    ch_pos = dict(zip(ch_names, elec_locs))
    meg_montage = mne.channels.make_dig_montage(ch_pos=ch_pos,
                                                coord_frame='head')
```

It seems to me like each channel should have a position that is settable. So I probably didn't get parts of mne-tools/mne-python#9447

I am furthermore confused that **part** of the warning is telling me that positions for some EEG channels could not be set, so I should consider setting the channel type to EEG. :thinking:

Reply:
I think I see the confustion -- the real issue here is that MEG (or MISC, or EOG, etc.) channel positions are not actually / cannot be set using `set_montage`, only EEG-like channels can be. So it seems like you are doing something wrong, but the warning is perhaps unclear.

Reply:
(I'm pretty sure this was the behavior even in 0.23 and before, I just noticed it and made it a warning for people in case they thought they could set MEG channel positions with `set_montage`...)

Reply:
In other words, I think the correct fix here is not to expect or use `set_montage` to set channel positions for anything other than EEG/sEEG/ECoG/DBS (or fNIRS, too, I think -- so that message should probably also be updated). Positions for other channel types like MEG should be set separately.

Reply:
Ahh, I see :thinking: it also says so in the docstring: https://mne.tools/stable/generated/mne.io.Raw.html?highlight=set_montage#mne.io.Raw.set_montage

although I think that docstring could maybe be extended (not only EEG chs work, apparently), and perhaps one could even add an RST warning to the docstring to make this more obvious?

Thanks! Still not sure how to fix our tests, but now it's clearer what to do.